modules/ROOT/pages/for-each-scope-concept.adoc

```diff
@@ -22,6 +22,37 @@ Note that if the input contains information outside the collection you tell it t
 
 You can also split an array in batches. Potentially, batches produce quicker processing. Each batch is treated as a separate Mule message. For example, if a collection has 200 elements and you set Batch Size (`batchSize`) to `50`, the For Each scope iteratively processes 4 batches of 50 elements, each as a separate Mule message.
 
+== Variable Propagation
+
+Every execution of the ForEach scope will start with the same variables and values as before the execution of the block. New variables or modifications of already existing ones while processing one element will not be visible while processing another element. All of those variable changes will not be avaiable outside the ForEach scope. In other words, the set of variables (and their values) after the execution of the ForEach Scope will remain the same as before the execution of it.
+
+[source,xml,linenums]
+----
+<set-variable variableName="var1" value="var1"/>
+<set-variable variableName="var2" value="var2"/>
+<foreach #[['apple', 'banana', 'orange']]>
+    <choice>
+        <when expression="#[payload == 'apple']">
+            <set-variable variableName="var2" value="newValue"/>
+            <set-variable variableName="var3" value="appleVal"/>
+        </when>
+        <when expression="#[payload == 'banana']">
+            <set-variable variableName="var3" value="bananaVal"/>
+        </when>
+        <otherwise>
+            <set-variable variableName="var3" value="otherVal"/>
+            <set-variable variableName="var4" value="val4"/>
+        </otherwise>
+    </choice>
+</foreach>
+----
+
+After aggregation, the variables are:
+
+`{var1: "var1", var2: "var2"}`
+
+None of the modifications are registered, including the creation of new variables. 
```

Comment:
`registered` -> `propagated`? registered sounds like they won't work even work in the foreach iteration scope

Reply:
```suggestion
None of the modifications are propagated, including the creation of new variables. 
```